docs/src/main/docs/guides/12_tracing_mp_guide.adoc

```diff
@@ -0,0 +1,764 @@
+----
+@ApplicationScoped
+public class GreetingProvider {
+    private final AtomicReference<String> message = new AtomicReference<>();
+
+    @Inject
+    public GreetingProvider(@ConfigProperty(name = "app.greeting") String message) {
+        this.message.set(message);
+    }
+
+    @Traced // <1>
+    String getMessage() {
+        return getMessage2();
+    }
+
+    @Traced  // <2>
+    String getMessage2() {
+        return message.get();
+    }
+
+    void setMessage(String message) {
+        this.message.set(message);
+    }
+}
+----
+<1> The `getMessage` method will be traced since it is externally invoked by `GreetResource`.
+<2> The `getMessage2` method will not be traced, even with the @Traced annotation, since it is called internally by `getMessage`.
+
+[source,bash]
+.Build and run the application, then invoke the endpoints and check the response:
+----
+curl http://localhost:8080/greet
+curl -X GET "http://localhost:9411/api/v2/spans?serviceName=helidon-mp-1" -H "accept: application/json"
+...
+[
+...
+  "io.helidon.examples.quickstart.mp.greetingprovider.getmessage", // <1>
+...
+]
+----
+<1> The `getMessage` method is traced, but `getMessage2` is not.
+
+=== Tracing across services
+
+Helidon automatically traces across services, providing that the services use the same tracer, for example, the same instance of Zipkin.
+This means a single trace can include spans from multiple services and hosts.  OpenTracing uses a `SpanContext` to
+propagate tracing information across process boundaries.  When you make client API calls, Helidon will
+internally call OpenTracing APIs to propagate the `SpanContext`. There is nothing you need to do in your application to make this work.
+
+To demonstrate distributed tracing, you will need to create a second project, where the server listens on port 8081.
+Create a new root directory to hold this new project, then do the following steps, similar to
+what you did at the start of this guide:
+
+==== Create a second service
+
+[source,bash,subs="attributes+"]
+.Run the Maven archetype:
+----
+mvn archetype:generate -DinteractiveMode=false \
+    -DarchetypeGroupId=io.helidon.archetypes \
+    -DarchetypeArtifactId=helidon-quickstart-mp \
+    -DarchetypeVersion={helidon-version} \
+    -DgroupId=io.helidon.examples \
+    -DartifactId=helidon-quickstart-mp-2 \
+    -Dpackage=io.helidon.examples.quickstart.mp
+----
+
+[source,bash]
+.The project will be built and run from the `helidon-quickstart-mp` directory:
+----
+cd helidon-quickstart-mp-2
+----
+
+[source,xml]
+.Add the following dependency to `pom.xml`:
+----
+<dependency>
+  <groupId>io.helidon.tracing</groupId>
+  <artifactId>helidon-tracing-zipkin</artifactId>
+</dependency>
+----
+
+[source,bash]
+.Replace `META-INF/microprofile-config.properties` with the following:
+----
+app.greeting=Hello From MP-2
+tracing.service=helidon-mp-2
+
+# Microprofile server properties
+server.port=8081
```

Comment:
On my Mac port 8081 is in use by a McAfee agent (I hit this on the SE guide as well). So I used 8085.  No problem -- I just don't know how common this would be. And whose to say 8085 is any better. I'm OK leaving it as is.

Reply:
ok I have Mcafee on my Mac and don't see that problem.